src/Kestrel.Core/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -549,5 +516,14 @@ private sealed class UTF8EncodingSealed : UTF8Encoding
 
             public override byte[] GetPreamble() => Array.Empty<byte>();
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal unsafe static Span<byte> UnsafeAsSpan(this ref ReadOnlySpan<byte> span)
```

Comment:
Just change the parser interface to take ROSpan

Reply:
I can, but I'll have to push this lower. Eventually the span gets rewritten in place. :/

Reply:
Changed.